framework/db/BatchQueryResult.php

```diff
@@ -214,8 +214,11 @@ private function getDbDriverName()
             return $this->db->driverName;
         }
 
-        if (isset($this->_batch[0]->db->driverName)) {
-            return $this->_batch[0]->db->driverName;
+        if (isset($this->_batch) && is_array($this->_batch) && !empty($this->_batch)) {
```

Comment:
`isset($this->_batch)` is redundant (`!empty($this->_batch)` already ensures that). `is_array($this->_batch)` probably too - `$this->_batch` is either `null` or `array`, so simple `!empty($this->_batch)` should be enough.